lib/cn/csched_sp3_work.c

```diff
@@ -774,7 +776,7 @@ sp3_work_wtype_length(
         *action = CN_ACTION_COMPACT_K;
         *rule = CN_RULE_LENGTH_MIN;
 
-        if (!atomic_read(&tn->tn_readers)) {
+        if (!rp->csched_full_compact && !atomic_read(&tn->tn_readers)) {
```

Comment:
IMO, we shouldn't even do k-compacts when doing full-compaction, it should just be a bunch of kv-compactions triggered via the garbage rule.  Should take much less code and only need to set the gc trigger low...  So I don't really understand why all this extra stuff going on...

Reply:
i.e., the csched_gc_pct rparam set to 1 or some such should be able to trigger a full gc on every node, reducing them all to one kvset via kv-compact.  It also then waits for all root spills to finish before doing the nodes in the case where more of the data is in the root than in the leaves...

Reply:
another advantage is that it will compact nodes in order of nodes with most garbaage first, so if you do interrupt it at some point it will have cleaned up all the nodes with the most garbge up to that point...